drape/overlay_tree.cpp

```diff
@@ -153,13 +153,21 @@ void OverlayTree::StartOverlayPlacing(ScreenBase const & screen, int zoomLevel)
   m_zoomLevel = zoomLevel;
 }
 
-void OverlayTree::Remove(ref_ptr<OverlayHandle> handle)
+bool OverlayTree::Remove(ref_ptr<OverlayHandle> handle)
 {
   if (m_frameCounter == kInvalidFrame)
-    return;
+  {
+    if (!IsEmpty())
+      Clear();
+    return true;
```

Comment:
Hm, Is it a valid state or something went wrong and we made this patch? When
```
m_frameCounter == kInvalidFrame && !IsEmpty()
```

Reply:
Yes, there are several places where `m_frameCounter` setted to  `kInvalidFrame` without call of `Clear()`, for example `OverlayTree::SetVisualScale`.
And this is ok, because of only `OverlayTree::Remove` call can cause dangling `ref_ptr`. 
So with something like map scale change and move we can get kInvalidFrame via `OverlayTree::SetVisualScale` and only then call to `OverlayTree::Remove` before start of next frame. And in that case we get danging `ref_ptr` without extra `Clear` call.